### DISCUSSION TITLE
Disabling sound/audio completely

### DISCUSSION THREAD

**Discussion by fragtion** [LizardByte/community#70] [Sunshine]
### Describe the Bug

I'm using Sunshine+Moonlight for remote desktop purposes, and would like to disable audio entirely (not just mute it on the client side) so as to make the session more bandwidth efficient, as I don't always require audio.  Can this be achieved with the current version, or will it need to be implemented as a configurable option? I know it's possible to choose the sound device, but I don't see how to disable it entirely. Thanks !

### Operating System Version

11

### Architecture

64-bit

### Sunshine Version

latest

### GPU Type

Nvidia

### GPU Model

GTX 1060

### GPU Driver/Mesa Version

latest

**Comment by jdfr228** [comment]
Setting up my Wii U GamePad to act as a second monitor for DS/3DS emulation after seeing a Steam Deck article of someone doing it with theirs, and I would also benefit from disabling audio encoding/streaming to save on any bandwidth possible!

**Comment by ReenigneArcher** [comment]
I'll leave this open, but I doubt anyone will work on this feature. You'd save a few kbps which is a drop in the ocean compared to how much video data is sent.

If you're that worried about saving bandwidth then:

A. Lower your resolution. Even one setting lower has a significant impact.
B. Lower the bitrate in moonlight.

**Reply by jdfr228** [reply]
Figured audio bitrate was at least a few hundred kb/s. The lowest quality mp3s hit 96kb/s and CDs sit at around 1100. I was thinking maybe 320kb/s were for audio, not factoring in surround sound support.

Do you happen to know typical audio bitrates for Sunshine? Does it vary by resolution or FPS? I'm limited to about 4000kb/s on my crappy network and a few hundred kb gained could make a big difference.

Of course, none of that matters if no one is interested in implementing it- Sunshine's primary purpose is game streaming after all.

**Reply by ReenigneArcher** [reply]
> Do you happen to know typical audio bitrates for Sunshine?

Sorry, I do not know.

> Does it vary by resolution or FPS?

Doubtful since those are not audio settings.

Are you even measuring how much bandwidth Sunshine is using?

**Reply by jdfr228** [reply]
Youtube will scale audio bitrate with video resolution so that's not unheard of, and doubling the FPS I would imagine would effectively double the audio bitrate since twice as many frames are delivered each second, but maybe audio is decoupled from video so that doesn't matter.  It might all depend on a lot of factors too, such as encoder/decoder settings.

I was not aware of a simple way to measure the bandwidth Sunshine was using, what would you recommend?

**Reply by ReenigneArcher** [reply]
Audio and video are sent over different ports.

I guess you could always block the audio port in the firewall. The ports are documented here: https://lizardbyte--330.org.readthedocs.build/projects/sunshine/en/330/about/advanced_usage.html#port

I don't have any suggestion for measuring bandwidth, but it seems you probably want to start with that. There's probably some settings you can change on the host that will reduce the data sent.

**Comment by Jamesits** [comment]
I am in a situation where having Sunshine capturing the sound actively made things worse.

I have a complex sound processing pipeline (for ASIO: virtual sound card -> VST hooks -> ASIO -> sound card; for system APIs: APO -> VST hooks -> sound card) on the Sunshine server. When Sunshine tries to capture the sound on my sound card (whether using `audio_sink` or `virtual_sink`), it introduces a lot jitter which resulted in random noises or frequency shifts on the audio I hear.

I can streaming the audio in a better way (there is a physical audio cable between the 2 computers). In this case, is there any way to make Sunshine not touch the audio devices?

**Reply by Jamesits** [reply]
Found a workaround: set `audio_sink` and `virtual_sink` to something that does not exist. Works on Sunshine Windows 0.22.0
